JS/Lesson19/goksel/recursion.js

```diff
@@ -36,103 +36,231 @@ console.log('5', getFactorial(5));
-		return;
-	} else {
-		console.log(arr[0]); // prints 1
-		return printingOutOneOfTheElements(arr.slice(1)); // printingOutOneOfTheElements([2,3,4,5]);
-	}
+    if(arr.length == 0){
+        return;
+    } else {
+        console.log(arr[0]); // prints 1
+        return printingOutOneOfTheElements(arr.slice(1)); // printingOutOneOfTheElements([2,3,4,5]);
+    }
 };
 printingOutOneOfTheElements(arrEx1);
 
+
+console.log("---Ex2---");
+
 // 2. Write a recursive function to capitalize the first letter of each word in a sentence.
 const ex2sentence =
-	'a recursive function to capitalize the first letter of each word in a sentence.';
+    'a recursive function to capitalize the first letter of each word in a sentence.';
 
 function capitalizeFirstLetter (sentence) {
-	const sentenceArray = sentence.split(' ');
-	if(sentenceArray.length === 0 || sentenceArray[0].length === 0) {
-		return '';
-	}
-	else {
-		const firstWord = sentenceArray[0];
-		const firstWordCapitalized = firstWord[0].toUpperCase() + firstWord.slice(1);
-		const restOfTheSentenceArray = sentenceArray.slice(1);
-		const restOfTheSentence = restOfTheSentenceArray.join(' ');
-		return firstWordCapitalized + ' ' + capitalizeFirstLetter(restOfTheSentence) ;
-	} 
+    const sentenceArray = sentence.split(' ');
+    if(sentenceArray.length === 0 || sentenceArray[0].length === 0) {
+        return '';
+    }
+    else {
+        const firstWord = sentenceArray[0];
+        const firstWordCapitalized = firstWord[0].toUpperCase() + firstWord.slice(1);
+        const restOfTheSentenceArray = sentenceArray.slice(1);
+        const restOfTheSentence = restOfTheSentenceArray.join(' ');
+        return firstWordCapitalized + ' ' + capitalizeFirstLetter(restOfTheSentence) ;
+    } 
 }
 
 console.log(capitalizeFirstLetter(ex2sentence));
 
 
+
+console.log("---Ex3---");
 // 3. Write a recursive function to count the number of vowels in a string.
 const sentenceEx4 = 'Hello, how are you?';
 const vowels = ['a', 'e', 'i', 'o', 'u'];
 
 function vowelsInString(sentence){
-	if (sentence.length === 0){ 
-		return 0;
-	} else {
-		if(vowels.includes(sentence[0].toLowerCase())){
-			return 1 + vowelsInString(sentence.slice(1));
-		} else {
-			return vowelsInString(sentence.slice(1));
-		}
-	}
+    if (sentence.length === 0){ 
+        return 0;
+    } else {
+        if(vowels.includes(sentence[0].toLowerCase())){
+            return 1 + vowelsInString(sentence.slice(1));
+        } else {
+            return vowelsInString(sentence.slice(1));
+        }
+    }
 }
 console.log(vowelsInString(sentenceEx4));
+
+
+
+console.log("---Ex4---");
 // 4. Write a recursive function to remove all occurrences of a specified character from a string.
 const sentenceEx5 = 'Hello, how are you?';
 const charToRemove = 'o';
 
+function removeCharacter(sentence, removeWord){
+    if(sentence.length === 0 || sentence[0].length === 0){
+        return "";
+        }
+        if(sentence[0] === removeWord ){
```

Comment:
```suggestion
        if(sentence[0] === characterToRemove ){
```

Reply:
I wrote removeWord but ı put characterToRemove when ı  log the function. But okay no need to add extra value there ı can use the original variable ın function.